plugins/woocommerce/src/Internal/ProductQueryFilters/FilterClausesGenerator.php

```diff
@@ -0,0 +1,377 @@
+	 * @param \WP_Query $wp_query WP_Query object.
+	 * @return array
+	 */
+	public function add_query_clauses( $args, $wp_query ) {
+		// Rating filter is handled by tax query.
+		if ( $wp_query->get( 'filter_stock_status' ) ) {
+			$stock_statuses = trim( $wp_query->get( 'filter_stock_status' ) );
+			$stock_statuses = explode( ',', $stock_statuses );
+			$stock_statuses = array_filter( $stock_statuses );
+
+			$args = $this->add_stock_clauses( $args, $stock_statuses );
+		}
+
+		if ( $wp_query->get( 'min_price' ) || $wp_query->get( 'max_price' ) ) {
+			$price_range = array(
+				'min_price' => $wp_query->get( 'min_price' ),
+				'max_price' => $wp_query->get( 'max_price' ),
+			);
+			$price_range = array_filter( $price_range );
+			$args        = $this->add_price_clauses( $args, $price_range );
+		}
+
+		$args = $this->add_attribute_clauses(
+			$args,
+			$this->get_chosen_attributes( $wp_query->query_vars )
+		);
+
+		return $args;
+	}
+
+	/**
+	 * Add query clauses for stock filter.
+	 *
+	 * @param array $args           Query args.
+	 * @param array $stock_statuses Stock statuses to be queried.
+	 * @return array
+	 */
+	public function add_stock_clauses( $args, $stock_statuses ) {
+		if ( empty( $stock_statuses ) ) {
+			return $args;
+		}
+
+		$args['join']   = $this->append_product_sorting_table_join( $args['join'] );
+		$args['where'] .= ' AND wc_product_meta_lookup.stock_status IN ("' . implode( '","', array_map( 'esc_sql', $stock_statuses ) ) . '")';
+
+		return $args;
+	}
+
+	/**
+	 * Add query clauses for price filter.
+	 *
+	 * @param array $args        Query args.
+	 * @param array $price_range {
+	 *     Price range array.
+	 *
+	 *     @type int|string $min_price Optional. Min price.
+	 *     @type int|string $max_price Optional. Max Price.
+	 * }
+	 * @return array
+	 */
+	public function add_price_clauses( $args, $price_range ) {
+		if ( ! isset( $price_range['min_price'] ) && ! isset( $price_range['max_price'] ) ) {
+			return $args;
+		}
+
+		global $wpdb;
+
+		$adjust_for_taxes = $this->adjust_price_filters_for_displayed_taxes();
+		$args['join']     = $this->append_product_sorting_table_join( $args['join'] );
+
+		if ( isset( $price_range['min_price'] ) ) {
+			$min_price_filter = intval( $price_range['min_price'] );
+
+			if ( $adjust_for_taxes ) {
+				$args['where'] .= $this->get_price_filter_query_for_displayed_taxes( $min_price_filter, 'max_price', '>=' );
+			} else {
+				$args['where'] .= $wpdb->prepare( ' AND wc_product_meta_lookup.max_price >= %f ', $min_price_filter );
+			}
+		}
+
+		if ( isset( $price_range['max_price'] ) ) {
+			$max_price_filter = intval( $price_range['max_price'] );
+
+			if ( $adjust_for_taxes ) {
+				$args['where'] .= $this->get_price_filter_query_for_displayed_taxes( $max_price_filter, 'min_price', '<=' );
+			} else {
+				$args['where'] .= $wpdb->prepare( ' AND wc_product_meta_lookup.min_price <= %f ', $max_price_filter );
+			}
+		}
+
+		return $args;
+	}
```

Comment:
@imanish003 @kmanijak, for example, we can use this method for price rang filter https://github.com/woocommerce/woocommerce/pull/42858